buildSrc/src/main/kotlin/com/sourceforgery/tachikoma/buildsrc/KotlinSetup.kt

```diff
@@ -0,0 +1,79 @@
+package com.sourceforgery.tachikoma.buildsrc
+
+import com.sourceforgery.tachikoma.buildsrc.CheckDuplicateClassesTask
+import hk2Version
+import implementation
+import kotlinVersion
+import org.gradle.api.Project
+import org.gradle.api.tasks.bundling.Jar
+import org.gradle.kotlin.dsl.apply
+import org.gradle.kotlin.dsl.dependencies
+import org.gradle.kotlin.dsl.get
+import org.gradle.kotlin.dsl.getByType
+import org.gradle.kotlin.dsl.getValue
+import org.gradle.kotlin.dsl.getting
+import org.gradle.kotlin.dsl.provideDelegate
+import org.gradle.kotlin.dsl.registering
+import org.gradle.kotlin.dsl.withType
+import org.jetbrains.dokka.gradle.DokkaTask
+import org.jetbrains.kotlin.gradle.tasks.KotlinCompile
+import org.jlleitschuh.gradle.ktlint.KtlintExtension
+import sourceSets
+import spekVersion
+import testImplementation
+
+fun Project.kotlinSetup() {
+    apply(plugin = "kotlin")
+    apply(plugin = "org.jlleitschuh.gradle.ktlint")
+    apply(plugin = "org.jetbrains.dokka")
+
+    dependencies {
+        testImplementation("org.jetbrains.spek:spek-junit-platform-engine:$spekVersion")
+        testImplementation("org.jetbrains.kotlin:kotlin-test:$kotlinVersion")
+        testImplementation("org.junit.platform:junit-platform-runner:1.1.0")
+        testImplementation("org.glassfish.hk2:hk2-locator:$hk2Version")
+        testImplementation("org.glassfish.hk2:hk2-utils:$hk2Version")
+        implementation("org.jetbrains.kotlin:kotlin-stdlib-jdk8:$kotlinVersion")
+    }
+
+    tasks.withType<KotlinCompile>().configureEach {
+        kotlinOptions {
+            languageVersion = "1.2"
```

Comment:
why 1.2 and not latest (or whatever default is, 1.3?)?

Reply:
Because it used to be 1.2. I.e., no reason at all